gensim/models/poincare.py

```diff
@@ -0,0 +1,827 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Author: Jayant Jain <jayantjain1992@gmail.com>
+# Copyright (C) 2017 Radim Rehurek <me@radimrehurek.com>
+# Licensed under the GNU LGPL v2.1 - http://www.gnu.org/licenses/lgpl.html
+
+
+"""
+Python implementation of Poincare Embeddings, an embedding to capture hierarchical information,
+described in [1]
+
+This module allows training a Poincare Embedding from a training file containing relations from
+a transitive closure.
+
+
+.. [1] https://arxiv.org/pdf/1705.08039.pdf
+
+"""
+
+
+import csv
+import itertools
+import logging
+import os
+import random
+import time
+
+from autograd import numpy as np, grad
+from collections import defaultdict, Counter
+from numpy import random as np_random
+from smart_open import smart_open
+
+from gensim import utils
+from gensim.models.keyedvectors import KeyedVectors, Vocab
+from gensim.models.word2vec import Word2Vec
+
+logger = logging.getLogger(__name__)
+
+
+class PoincareModel(utils.SaveLoad):
+    """
+    Class for training, using and evaluating Poincare Embeddings described in https://arxiv.org/pdf/1705.08039.pdf
+
+    The model can be stored/loaded via its `save()` and `load()` methods, or stored/loaded in the word2vec
+    format via `wv.save_word2vec_format()` and `KeyedVectors.load_word2vec_format()`.
+
+    Note that training cannot be resumed from a model loaded via `load_word2vec_format`, if you wish to train further,
+    use `save()` and `load()` methods instead.
+
```

Comment:
`<nitpick>`
You're inconsistent about empty lines. Sometime you leave the last line in the docstring empty; other times you leave the first line after the docstring empty; other times none of the two. Ordnung muss sein! :) 
`</nitpick>`